Using NOCOPY code relocation generates a warning flag

**Describe the bug**
When relocating code to external flash for XIP, the NOCOPY flag is used. This prevents the code from being copied to RAM. Doing this means the functin `void data_copy_xip_relocation(void)` is left empty, since no copying has to be done. However the following is generated:

```
/* Auto generated code. Do not modify.*/
#include <zephyr/kernel.h>
#include <zephyr/linker/linker-defs.h>
#include <zephyr/kernel_structs.h>
#include <kernel_internal.h>

void data_copy_xip_relocation(void)
{
void;
}

void bss_zeroing_relocation(void)
{
return;
}

```

This code generates the following warning: 
  ```
  [164/179] Building C object zephyr/CMakeFiles/code_relocation_source_lib.dir/code_relocation.c.obj
    zephyr/code_relocation.c: In function 'data_copy_xip_relocation':
    zephyr/code_relocation.c:10:1: warning: useless type name in empty declaration
       10 | void;
          | ^~~~
```

**To Reproduce**
Steps to reproduce the behavior:

- Comment out the following lines in CMakeLists.txt in `samples/application_development/code_relocation_nocopy/` and leave the line which places TEXT section in external flash with the NOCOPY flag.
```
# # But still relocate (copy) the data to RAM
zephyr_code_relocate(FILES src/ext_code.c LOCATION RAM_DATA)

# # sram_code instead runs entirely from SRAM after being copied there.
zephyr_code_relocate(FILES src/sram_code.c LOCATION RAM)
```

- run `west build -p always -b nrf5340dk_nrf5340_cpuapp zephyr/samples/application_development/code_relocation_nocopy/`

- warning will output right before linking stage

**Expected behavior**
The sample should build without any warnings. The `void` keyword in the generated `data_copy_xip_relocation` file should be replaced by a `return` keyword.

**Impact**
Although functionality this does not break anything, it is not nice to start of the development of an application with a warning that you can't remove. Suppressing the warning through CMake could be possible but not as straightforward. 

**Environment (please complete the following information):**

    OS: Ubuntu 20.04 (https://github.com/zephyrproject-rtos/docker-image)
    Toolchain: Zephyr SDK 0.15.1
    Commit: [e9a5951](https://github.com/zephyrproject-rtos/zephyr/commit/e9a59518fcbd54f2ee27ea3f190a8fe7bd688f35)

--- a/scripts/build/gen_relocate_app.py
+++ b/scripts/build/gen_relocate_app.py
@@ -392,7 +392,7 @@ def dump_header_file(header_file, code_generation):
     if code_generation["copy_code"]:
         code_string += DATA_COPY_FUNCTION.format(code_generation["copy_code"])
     else:
-        code_string += DATA_COPY_FUNCTION.format("void;")
+        code_string += DATA_COPY_FUNCTION.format("return;")
     if code_generation["zero_code"]:
         code_string += BSS_ZEROING_FUNCTION.format(code_generation["zero_code"])
     else:

